Gems/AtomLyIntegration/AtomFont/Code/Source/FFont.cpp

```diff
@@ -1486,9 +1486,10 @@ bool AZ::FFont::UpdateTexture()
     imageUpdateReq.m_sourceData = m_fontTexture->GetBuffer();
     imageUpdateReq.m_sourceSubresourceLayout = layout;
 
-    m_fontStreamingImage->UpdateImageContents(imageUpdateReq);
+    const RHI::ResultCode result = m_fontStreamingImage->UpdateImageContents(imageUpdateReq);
+    AZ_Error("FFont", result == RHI::ResultCode::Success, "Cannot update font texture, code %d", (unsigned int)result);
```

Comment:
I would suggest to recreate the m_fontStreamingImage by using CreateFromCpuData function, instead of using UpdateImageContents function. It shouldn't effect performance much if this function won't be called frequently.
Or you can switch to use AZ::Data::Instance<AZ::RPI::AttachmentImage> for m_fontStreamingImage which supports UpdateImageContents 

Neither of them need change Atom's RHI for the support.

Reply:
Update content is called frequently on UI dialog initialization. That code that saves each new texture produced 24 images for one particular UI dialog. By design the update happens after each string output, not once per frame (mark as dirty multiple times, update once). But once a UI dialog initialization complete it should be fine if we have large enough buffer.
I'll try re-creating the image instead of the update.

Reply:
AZ::RPI::AttachmentImage may be better choice if the update happens frequently. CreateFromCpuData triggers gpu memory allocation for the new image it creates. Although if the image size is small, then it may not be noticeable.

Reply:
Our current font size is typical 1024x512, I believe we need twice that 1024x1024.

Reply:
I was actually wrong about texture update frequency, I added the texture save-to-file code piece on texture update, not GPU image update. So it should be fine, GPU image is updated just once per an UI dialog I believe.